tests/components/faadelays/test_config_flow.py

```diff
@@ -0,0 +1,104 @@
+
+
+async def mock_valid_airport(self, *args, **kwargs):
+    """Return a valid airport."""
+    self.name = "Test airport"
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch.object(faadelays.Airport, "update", new=mock_valid_airport), patch(
+        "homeassistant.components.faadelays.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.faadelays.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "id": "test",
+            },
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Test airport"
+    assert result2["data"] == {
+        "id": "test",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_airport(hass):
+    """Test we handle invalid airport."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "faadelays.Airport.update",
+        side_effect=faadelays.InvalidAirport,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "id": "test",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {CONF_ID: "invalid_airport"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle a connection error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("faadelays.Airport.update", side_effect=ClientConnectionError):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "id": "test",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unexpected_exception(hass):
+    """Test we handle an unexpected exception."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("faadelays.Airport.update", side_effect=HomeAssistantError):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "id": "test",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
```

Comment:
Please also test an already created entry with the same id as the one entered in the flow, which should abort the flow as already configured. Add a `MockConfigentry` before starting the flow.

https://github.com/home-assistant/core/blob/3bdf9628385a381d28f1ee092ab686eed23f8163/tests/common.py#L656